domain-models-maven-plugin/src/main/java/org/folio/rest/tools/ClientGenerator.java

```diff
@@ -334,13 +334,13 @@ public void generateMethodMeta(String methodName, JsonObject params, String url,
     /* Handle place holders in the URL
       * replace {varName} with "+varName+" so that it will be replaced
       * in the url at runtime with the correct values */
-    Matcher m = Pattern.compile("\\{.*?\\}").matcher(url);
-    while(m.find()){
+    Matcher m = Pattern.compile("\\{.*?\\}").matcher("^" + url);
+    while (m.find()) {
       String varName = m.group().replace("{","").replace("}", "");
       url = url.replace("{"+varName+"}", "\"+"+varName+"+\"");
     }
 
-    url = "\""+url.substring(1)+"\"+queryParams.toString()";
+    url = "\"" + url + "\"+queryParams.toString()";
```

Comment:
```suggestion
    url = "\"" + 
        Pattern.compile("\\{([^{}/]+)\\}").matcher(url).replaceAll("\"+$1+\"") + 
        "\"+queryParams.toString()";
```

Reply:
This is supposed to replace the preceding for loop.. That results in failures.. We need a better test for clientGenerator itself --  the TestResource.java does not expose enough features!

Reply:
This should replace all the other Matcher and url code above.

Reply:
OK.. I'll try again.. I like the simplication.

Reply:
OK.. I'll try again.. I like the simplification.

Reply:
OK. Good now.